### ISSUE TITLE
elasticsearch_kibana_alert broken for versions above v7.13?

### ISSUE THREAD

**Issue by nbs-rsutton** [phillbaker/terraform-provider-elasticsearch#253]
Use of /api/alerts/alert/ endpoint deprecated since v7.13 - https://www.elastic.co/guide/en/kibana/current/alerts-api.html

It appears that elasticsearch_kibana_alert refers to this deprecated endpoint, as opposed to /api/alerting/rule/ for creating rules in kibana.

When attempting to prove the functionality of this resource using the example, the following error is returned:
Error: no active connection found: no Elasticsearch node available

If this provider is valid for use between 7.7 and 7.13 then please let me know and I'll raise a PR to add this to the documentation.

I am not currently up to speed with Golang otherwise I'd take a deeper look...
Would appreciate if someone more experienced would be able to take a look?

**Comment by phillbaker** [comment]
Hello - the endpoint is deprecated, meaning it will be removed in ESv8, however it still works in all versions of ESv7.

>  Error: no active connection found: no Elasticsearch node available

This is because you couldn't connect to the Elasticsearch cluster. Please ensure you can connect to ES from wherever you're running terraform. Otherwise, please include the following:
* terraform version
* provider version
* elasticsearch version (and opendistro version if relevant)
* redacted version of the terraform resource configuration
* terraform logs by setting [`TF_LOG=info`](https://www.terraform.io/docs/internals/debugging.html)

**Comment by nbs-rsutton** [comment]
Am going to close this issue:

**Error: no active connection found: no Elasticsearch node available**

Initial error was being caused (as you advised!), by the endpoint being pulled being incorrect, this was pointed at the elastic endpoint and not the kibana one.

**Error: invalid character "<" looking for beginning of value**

Second error being seen when testing locally (as above) was being caused due to the resource requiring both the elastic url (url) and kibana url needing to be specified (kibana_url) within the provider (I noticed this in the provider doc retrospectively which was unfortunate!)

Not sure if it's possible, but it may be worth adding some extra error handling into the resource for this? (e.g. should the kibana_url not have been set in the provider, throw if user invoking kibana resource?) Just to make this a bit clearer should anyone run into this in the future. I think the second error was being caused by it expecting json but receiving a html 403 which adds extra layers of digging to understand the root of the issue? Just a suggestion though 👍

Appreciate the guidance on this one, apologies if I reached out a bit early into investigations, my head had begun hurting from repetitive high fives with my desk when I sent the initial query through... just happy it's been resolved now!
